shared/login/signup/invite-code.render.desktop.js

```diff
@@ -3,36 +3,68 @@
 
 import React, {Component} from 'react'
 import {globalStyles} from '../../styles/style-guide'
-import {Text, Input, Button} from '../../common-adapters'
-
+import {Text, Input, Button, Icon} from '../../common-adapters'
+import Container from '../forms/container'
 import type {Props} from './invite-code.render'
 
 export default class Render extends Component {
   props: Props;
 
+  state: {
+    inviteCode: ?string
+  };
+
+  constructor (props: Props) {
+    super(props)
+    this.state = {
+      inviteCode: this.props.inviteCode || ''
+    }
+  }
+
   render () {
-    let inviteRef = null
     const submitInviteCode = () => {
-      const inviteCode = inviteRef && inviteRef.getValue() || ''
-      this.props.onInviteCodeSubmit(inviteCode)
+      this.props.onInviteCodeSubmit(this.state.inviteCode)
+      this.setState({inviteCode: ''})
```

Comment:
Thanks, removed.  The point was to disable the submit button after submitting an invalid signup code until you make a change to the text field (`disabled={!this.state.inviteCode}`), but I don't like it after all.